opentracing-impl/src/main/java/io/opentracing/propagation/TextMapImpl.java

```diff
@@ -16,7 +16,7 @@
 import java.util.Iterator;
 import java.util.Map;
 
-public final class TextMapImpl implements TextMapWriter, TextMapReader {
+public final class TextMapImpl implements TextMap {
```

Comment:
I would rather keep TextMapWriter and TextMapReader (instead of Impl) as concrete implementations, and let them throw NotImplementedException from one of the methods.

It's also problematic that this class is defined in the `-impl` jar, because it is intended for the end user of the API, and the end users have no need to depend on the `-impl` module.